src/commands/sourcemaps/git.ts

```diff
@@ -0,0 +1,242 @@
+}
+
+// Creates a lookup map from source paths to tracked file paths.
+// The tracked file paths will be split into as many keys as there is folders in the path (+ the filename).
+// It allows to do an exact match when looking up a source path in the tracked files, which will be enough
+// in most cases.
+// It also allows to make an 'educated guess' / 'best-effort' match on the off-chance the source path we are
+// trying to find does not contain the whole tracked file path. It's an edge case but can easily be handled
+// using this method.
+//
+// To perform a match simply lookup the source path in the map keys. The value returned is the complete tracked
+// file path.
+//
+// For example the following tracked file path:
+// project/folder1/folder2/src.js
+//
+// Will be matched by any of:
+// project/folder1/folder2/src.js
+// folder1/folder2/src.js
+// folder2/src.js
+// src.js
+//
+export const trackedFilesMap = (trackedFiles: string[]): Map<string, string> => {
+  const map = new Map<string, string>()
+  for (const trackedFile of trackedFiles) {
+    const split = trackedFile.split('/')
+    for (let i = 0; i < split.length; i++) {
+      map.set(split.slice(i, split.length).join('/'), trackedFile)
+    }
+  }
+
+  return map
+}
+
+export interface RepositoryData {
+  hash: string
+  remote: string
+  trackedFiles: Map<string, string>
+}
+
+// Gathers repository data.
+// It returns the current hash and remote as well as a map of tracked files.
+// Look for the trackedFilesMap function for more details.
+//
+// To obtain the list of tracked files path tied to specific sourcemaps, first invoke 'getRepositoryData',
+// then for each sourcemap invoke the 'filterTrackedFiles' function.
+export const getRepositoryData = async (
+  git: simpleGit.SimpleGit,
+  stdout: Writable,
+  repositoryURL: string | undefined
+): Promise<RepositoryData | undefined> => {
+  // Invoke git commands to retrieve the remote, hash and tracked files.
+  // We're using Promise.all instead of Promive.allSettled since we want to fail early if
+  // any of the promises fails.
+  let remote: string
+  let hash: string
+  let trackedFiles: string[]
+  try {
+    if (repositoryURL) {
+      ;[hash, trackedFiles] = await Promise.all([gitHash(git), gitTrackedFiles(git)])
+      remote = repositoryURL
+    } else {
+      ;[remote, hash, trackedFiles] = await Promise.all([gitRemote(git), gitHash(git), gitTrackedFiles(git)])
+    }
+  } catch (e) {
+    stdout.write(renderGitError(e))
+
+    return undefined
+  }
+
+  const data = {
+    hash,
+    remote,
+    trackedFiles: trackedFilesMap(trackedFiles),
+  }
+
+  return data
+}
+
+// Looks up the source paths declared in the sourcemap and try to match each of them
+// with a tracked file. The list of matching tracked files is returned.
+export const filterTrackedFiles = async (
+  stdout: Writable,
+  srcmapPath: string,
+  projectPath: string,
+  trackedFiles: Map<string, string>
+): Promise<string[] | undefined> => {
```

Comment:
How do we match tracked files with original source files in the source maps? Both are list, but they can be of different sizes, depending on the number of files that are matched, right?

Reply:
Yes they can be of different size. The idea I had in mind was to have a similar logic to match on the backend-side.

Reply:
What do you think about using a different data structure that allows us to skip matching? Gaving two places where we do matching seems error-prone. One way could be to represents files with no matches by empty strings, and consider the original source files array and this array always have the same size?

Reply:
I agree. The starting point of all this was the idea that we'd have the full list of tracked files available on the backend and do matches there. Allowing to apply any update to the logic to everyone instantaneously. 
Then came the idea of only sending only the strict minimal amount of tracked files since most of the tracked files (assets, ...) would be useless and sourcemaps only reference a small subset.
I'll have to think about it more thoroughly, I'll come up with something.

Reply:
It is now covered in e832d79 cc @lraucy 
As we talked about, we're now just trying to filter out unrelated tracked files.